go/src/lib.rs

```diff
@@ -327,62 +365,69 @@ fn convert_vec_to_pointer<T>(mut vec: Vec<T>) -> (*mut T, c_long) {
 }
 
 /// TODO: Avoid the use of expect and unwrap in the code and add a common error handling mechanism.
-fn valkey_value_to_command_response(value: Value) -> RedisResult<Option<CommandResponse>> {
+fn valkey_value_to_command_response(value: Value) -> RedisResult<CommandResponse> {
     let mut command_response = CommandResponse::default();
-    let result: RedisResult<Option<CommandResponse>> = match value {
-        Value::Nil => Ok(None),
+    let result: RedisResult<CommandResponse> = match value {
+        Value::Nil => Ok(command_response),
         Value::SimpleString(text) => {
             let vec: Vec<u8> = text.into_bytes();
             let (vec_ptr, len) = convert_vec_to_pointer(vec);
             command_response.string_value = vec_ptr as *mut c_char;
             command_response.string_value_len = len;
-            Ok(Some(command_response))
+            command_response.response_type = ResponseType::String;
+            Ok(command_response)
         }
         Value::BulkString(text) => {
             let (vec_ptr, len) = convert_vec_to_pointer(text);
             command_response.string_value = vec_ptr as *mut c_char;
             command_response.string_value_len = len;
-            Ok(Some(command_response))
+            command_response.response_type = ResponseType::String;
```

Comment:
Do we want to distinguish between two? (e.g. String and ByteArray)

Reply:
I don't currently see a need for that as both `String` and `ByteArray` will be handled in the same way in GO.